core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -571,71 +572,100 @@ public static Connection getDBConnection(DataSource dataSource) throws SQLExcept
     }
 
     public static void closeConnection(Connection dbConnection) {
-
-        if (dbConnection != null) {
-            try {
-                dbConnection.close();
-            } catch (SQLException e) {
-                log.error("Database error. Could not close statement. Continuing with others. - " + e.getMessage(), e);
-            }
-        }
+        close(dbConnection);
     }
 
-    private static void closeResultSet(ResultSet rs) {
-
-        if (rs != null) {
+    /**
+     *
+     * @param dbObject -  java.sql object types that implement both Wrapper and AutoCloseable interfaces
+     */
+    private static <AutoClosableWrapper extends AutoCloseable & Wrapper> void close(AutoClosableWrapper dbObject) {
+        if (dbObject != null) {
             try {
-                rs.close();
+                dbObject.close();
+            } catch (SQLRecoverableException ex) {
```

Comment:
Thank you for your review.   There is no need to change the code.  If the exception isn't thrown, the catch statement will not be entered, one of the other catch statements will handle it.  SQLException "should" be  thrown by every driver out there.  In the off-chance it doesn't, then every exception inherits from the Exception class.

https://docs.oracle.com/javase/tutorial/jdbc/basics/sqlexception.html#categorized_sqlexceptions